### ISSUE TITLE
Spotify is completely immune to MaxTo

### ISSUE THREAD

**Issue by vegardlarsen** [digitalcreations/MaxTo#211]
Spotify's latest release appears to be completely immune to any of MaxTo's functionality:

- maximize handler
- shift-dragging
- keyboard shortcuts

Figure out why.

**Comment by Radanz** [comment]
Agreed, used to work now does not, windows 10 latest updates and latest spotify.
Also get the same with the last few versions of Unity Game Editor. None of it windows are affected by MaxTo in any way.
Maybe they are using a higher level or newer path of Windows Manager? We all know hoe Microsoft loves to change things.
Hope its fixed soon, makes laying out my windows for programming on the 4k screen a tad annoying.
Anything I can do to help let me know.

**Comment by matthiask82** [comment]
Same problem here, came to report and saw this issue at the very top. Changed with Spotify's latest release.

**Comment by repentsinner** [comment]
Oof, just ran into this today as well with Spotify auto-update to `1.0.75.483.g7ff4a0dc` on Win 10 x64 version 1709 build 16299.248. Thanks for looking into this!

**Comment by natn** [comment]
I can confirm this behavior also.  One additional note to add:

If you hit maximize without moving the Spotify window it will snap into a region and can be moved between regions using keyboard shortcuts as with any other app, however if you drag or resize the the window using the mouse, it no longer responds to MaxTo. 

Spotify desktop version: 1.0.77.338.g758ebd78
Windows 10 Version:  1709 Build:  16299.309

**Comment by repentsinner** [comment]
Any luck on this? My MaxTo trial just ended, and I'm a bit hesitant to register until this is resolved as this Spotify is one of the five apps I use with it.

**Comment by vegardlarsen** [comment]
My initial examination of this shows that Spotify has simply stopped telling anyone else it is moving its window. I don't know why or how.

**Comment by repentsinner** [comment]
Thanks for the update @vegardlarsen! If there's anything else I can do to help diagnose or test from here, let me know.

**Comment by vegardlarsen** [comment]
Thank you all for reporting this. I have found a workaround, and will be shipping with this workaround enabled in the next release of MaxTo. To enable it now, you should be able to do the following:

1. Open up Notepad as Administrator (or any other text editor, Start, type `notepad`, right click and select **Run as administrator**)
2. Open the `shims.xml` file in `C:\Program files (x86)\MaxTo`
3. Under `<emulate_maximize>`, add this entry (win:

```xml
    <!-- Spotify -->
    <process title="Spotify">Spotify</process>
```

4. Restart MaxTo.
5. **Profit!**

Please let me know if this works for you.

**Comment by matthiash82** [comment]
Sorry to report that the workaround did *not* work for me. Fairly sure I
added the entry correctly in the XML-file. One difference however was that
I had MaxTo installed in "C:\Program Files", not "C:\Program Files (x86)".

**Comment by vegardlarsen** [comment]
That was my bad; I don't install MaxTo using the installer when I am developing, so I took the path from memory.

It shouldn't make a difference though; so now I am confused. 😕

**Comment by repentsinner** [comment]
@vegardlarsen any update on this, or any way we could help narrow it down?

**Comment by hensm** [comment]
The `<emulate_maximize>` workaround on its own didn't work for me. But I guess since Spotify is a Chromium instance, it shares the same window class as Chrome (`Chrome_WidgetWin_0`), which is in the ignore section. Removing that made Spotify work again. I don't know why it's there, but I don't use Chrome, so it's not a problem for me.

[shims.xml](https://gist.github.com/hensm/67aaed3881e5f12408e6af28b51e146a)

**Comment by vegardlarsen** [comment]
This has been fixed in the preview version published now (2.0.0-alpha.3). See #220 for information on how you can test the preview releases.
